screed/tests/test_attriberror.py

```diff
@@ -0,0 +1,85 @@
+import screed
+from screed.DBConstants import fileExtension
+import os
+
+
+class nostring:
+    def __init__(self):
+        self.exists = True
+
+    def __str__(self):
+        raise AttributeError
```

Comment:
Inappropriate yes but it seemed the easiest way to test the expected behavior. In theory, it should be impossible for an object to not have a __str__ function as it would be inherited from the Object class unless explicitly overridden (if I'm remembering my chat/hack session with @camillescott correctly).

Well, the goal of this pull was to fix that : 3

I think it's odd that screed doesn't throw errors when the isinstance fails. That seems like the most intuitive behavior to me. That said that's much more towards c++'s typed behavior and less like Python's dynamic types.